tests/Skills/Child/AdapterWithErrorHandler.cs

```diff
@@ -1,24 +1,93 @@
             : base(configuration, logger)
         {
-            OnTurnError = (turnContext, exception) =>
+            _conversationState = conversationState;
+            _userState = userState;
+
+            OnTurnError = async (turnContext, exception) =>
             {
                 // Log any leaked exception from the application.
-                logger.LogError($"Exception caught : {exception.Message}");
-                return Task.CompletedTask;
+                logger.LogError(exception, $"[OnTurnError] unhandled error : {exception.Message}");
+
+                // Send a message to the user
+                await turnContext.SendActivityAsync("The bot encounted an error or bug.");
+                await turnContext.SendActivityAsync("To continue to run this bot, please fix the bot source code.");
+
+                if (conversationState != null)
+                {
+                    try
+                    {
+                        // Delete the conversationState for the current conversation to prevent the
+                        // bot from getting stuck in a error-loop caused by being in a bad state.
+                        // ConversationState should be thought of as similar to "cookie-state" in a Web pages.
+                        await conversationState.DeleteAsync(turnContext);
+                    }
+                    catch (Exception e)
+                    {
+                        logger.LogError(e, $"Exception caught on attempting to Delete ConversationState : {e.Message}");
+                    }
+                }
+
+                // Send a trace activity, which will be displayed in the Bot Framework Emulator
+                //await turnContext.TraceActivityAsync("OnTurnError Trace", exception.Message, "https://www.botframework.com/schemas/error", "TurnError");
             };
         }
+
+        public override async Task<InvokeResponse> ProcessActivityAsync(ClaimsIdentity claimsIdentity, Activity activity, BotCallbackHandler callback, CancellationToken cancellationToken)
+        {
+            if (!claimsIdentity.Claims.Any(x => x.Type == "azp"))
+            {
+                if (_conversationMap.TryGetValue(activity.Conversation.Id, out string appId))
+                {
+                    claimsIdentity.AddClaim(new Claim("azp", appId));
+                    claimsIdentity.AddClaim(new Claim("ver", "2.0"));
+                }
+            }
+            else
+            {
+                var appId = JwtTokenValidation.GetAppIdFromClaims(claimsIdentity.Claims);
+                _conversationMap.AddOrUpdate(activity.Conversation.Id, appId, (id, a) => appId);
+            }
+
+            if (activity.Type == ActivityTypes.EndOfConversation)
+            {
+                _conversationMap.TryRemove(activity.Conversation.Id, out var conversation);
```

Comment:
why "out var conversation" if conversation is never used? 

Gabo recommends "_".